packages/checkbox/index.tsx

```diff
@@ -91,7 +94,10 @@ export class Checkbox extends React.Component<CheckboxProps, CheckboxState> {
       this.handleChange(checked!, indeterminate!);
     }
     if (disabled !== prevProps.disabled) {
-      this.foundation.setDisabled(disabled);
+      this.setState({
```

Comment:
@gugu I dont think we need to setState here. foundation.setDisabled() will call `setNativeControlDisabled`, which updates disabled for us.